services/storage/batch_cursor.go

```diff
@@ -0,0 +1,118 @@
+package storage
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/influxdata/influxdb/tsdb"
+)
+
+type singleValue struct {
+	v interface{}
+}
+
+func (v *singleValue) Value(key string) (interface{}, bool) {
+	return v.v, true
+}
+
+func newAggregateBatchCursor(ctx context.Context, agg *Aggregate, cursor tsdb.Cursor) tsdb.Cursor {
+	if cursor == nil {
+		return nil
+	}
+
+	switch agg.Type {
+	case AggregateTypeSum:
+		return newSumBatchCursor(cursor)
+	case AggregateTypeCount:
+		return newCountBatchCursor(cursor)
+
+	default:
+		// TODO(sgc): should be validated higher up
+		panic("invalid aggregate")
+	}
+}
+
+func newSumBatchCursor(cur tsdb.Cursor) tsdb.Cursor {
+	switch cur := cur.(type) {
+	case tsdb.FloatBatchCursor:
+		return &floatSumBatchCursor{FloatBatchCursor: cur}
+
+	case tsdb.IntegerBatchCursor:
+		return &integerSumBatchCursor{IntegerBatchCursor: cur}
+
+	case tsdb.UnsignedBatchCursor:
+		return &unsignedSumBatchCursor{UnsignedBatchCursor: cur}
+
```

Comment:
Extra blank line. In fact there are a few in `newCountBatchCursor` too.

Reply:
Blank lines were just for style, so it wasn't so densely packed, but happy to remove them.